### ISSUE TITLE
feat: Can the proxy-cache plugin support persistent caching?

### ISSUE THREAD

**Issue by w7team** [apache/apisix#10296]
### Description

我有一个想把第三方对象存储在各个cdn节点做持久缓存的需求，如果cdn节点能找到缓存文件就一致读本地的缓存，但是我看文档proxy-cache必须要设置的缓存时间，能否增加一个可以持久缓存的选项？

I have a requirement to store third-party objects in various CDN nodes for persistent caching. If a CDN node can find the cached file, it should consistently read the local cache. However, based on the documentation, it seems that the proxy-cache plugin requires setting a cache time. Is it possible to add an option for persistent caching?

**Comment by kayx23** [comment]
> it seems that the proxy-cache plugin requires setting a cache time. 

Note that `cache_ttl` is used with in-memory caching. You can cache on disk.

**Comment by kayx23** [comment]
Would adjusting the TTL to a large value not suffice?

**Comment by w7team** [comment]
有多个节点持久缓存的需求，但又不想每个节点一开始就全量同步所有文件，这样会给缓存节点的部署带来巨大的门槛。
设置一个足够大的值是一个方案，但我认为不是一个完美的方案，还是建议能有一个方案来从根本上解决这个问题。
我相信加了这个机制后，又给apisix增加了一个不可替代的优势。





------------------ Original ------------------
From: Traky Deng ***@***.***&gt;
Date: Tue,Oct 17,2023 1:19 PM
To: apache/apisix ***@***.***&gt;
Cc: 微擎团队 ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [apache/apisix] help request: Can the proxy-cache plugin supportpersistent caching? (Issue #10296)





 
Would adjusting the TTL to a large value not suffice?

**Comment by w7team** [comment]
这个会加入更新计划吗？

**Comment by kayx23** [comment]
cc @monkeyDluffy6017

**Comment by w7team** [comment]
@Revolyssup
@kingluo
@worldwonderer
@soulbird
@Chever-John
@agile6v 
几位大佬们抽空看看这个需求有必要做吗？

**Comment by monkeyDluffy6017** [comment]
Why do you need a cache that won't expire? Can't you set a big enough time for what you want?

**Comment by w7team** [comment]
> Why do you need a cache that won't expire? Can't you set a big enough time for what you want?

主要想利用proxy-cache 插件给对象存储服务做一个持久缓存，因为对象存储里的文件是长期不需要更新的，但是又不想全量复制到各个节点服务器上，这样会导致部署门槛变得非常高。如果设置一个非常大的过期值，也能解决，但无法从根本上解决这个问题，因为总有一天他还会从源站下载一次，这样就多消耗了一次源站的流量。如果能支持持久存储的参数不是更好嘛？

I mainly want to use the proxy-cache plug-in to create a persistent cache for the object storage service, because the files in the object storage do not need to be updated for a long time, but I do not want to copy them all to each node server, which will cause the deployment threshold to become very high. If you set a very large expiration value, it can be solved, but it cannot fundamentally solve the problem, because one day it will download from the origin site again, thus consuming one more traffic of the origin site. Wouldn't it be better if it could support persistent storage of parameters?

**Comment by monkeyDluffy6017** [comment]
I don't think this feature is necessary. APISIX is not a storage, it's only used to temporarily cache upstream responses

**Comment by w7team** [comment]
> @w7team Maybe you could check [proxy_store](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_store) if you need persistent storage of the response files from origin sites, although it's a static nginx config item. :-)

谢谢你的建议，我准备研究研究proxy_store

Thank you for your suggestion, I am going to study proxy_store

**Comment by monkeyDluffy6017** [comment]
considered resolved
